packages/astro/scripts/build.js

```diff
@@ -1,40 +1,85 @@
 import assert from 'node:assert';
-import { existsSync } from 'node:fs';
+import path from 'node:path';
+import { existsSync, rmSync, copyFileSync } from 'node:fs';
 import { cp, rm } from 'node:fs/promises';
 import { execa } from 'execa';
+import chokidar from 'chokidar';
 import esbuild from 'esbuild';
-import glob from 'fast-glob';
 import { nodeExternalsPlugin } from 'esbuild-node-externals';
 
-// clean dist
-await rm('dist', { recursive: true, force: true });
-
-// only do typechecking and emit the type declarations with tsc
-execa('tsc', ['--emitDeclarationOnly', '--project', './tsconfig.build.json'], {
-  stdio: 'inherit',
-  preferLocal: true,
-});
-
-// build with esbuild
-await esbuild.build({
-  entryPoints: ['src/index.ts'],
-  bundle: true,
-  tsconfig: './tsconfig.build.json',
-  platform: 'node',
-  format: 'esm',
-  outdir: 'dist',
-  define: {
-    'process.env.TUTORIALKIT_DEV': JSON.stringify(process.env.TUTORIALKIT_DEV ?? null),
-  },
-  plugins: [nodeExternalsPlugin()],
-});
-
-if (existsSync('./dist/default')) {
-  assert.fail('TypeScript transpiled the default folder, it means that the tsconfig has an issue');
+const isWatch = process.argv.includes('--watch');
+
+await generateTypes();
+await buildJS();
+await copyDefaultFolder();
+
+async function generateTypes() {
+  if (!isWatch) {
```

Comment:
Nit: I would always delete `dist`. It's too easy after a rename to have a file staying around and causing issues.

Reply:
In `pnpm` workspace this is required. When `--recursive --parallel` starts the scripts, we must have initial builds on file system ready. Otherwise the packages will try to build at the same time, and as their dependencies (other monorepo packages) are not present, they fail the build. https://egoist.dev/pnpm-monorepo

Reply:
Ah maybe we need something similar to what we've done in WebContainer then. In there, we wait for the `dist` folder to be ready before starting the `watch` tasks of the downstream packages.

For now, let's keep your logic then :+1: